source/magpie/group.d

```diff
@@ -317,6 +354,115 @@ public:
 
         return combine(pos);
     }
+
+    /++
+    Index operation of the form gp[size_t, size_t, size_t]
+    +/
+    auto opIndex(size_t i1, size_t i2, size_t i3)
+    {
+        static foreach(i; 0 .. GrpRowType.length)
+            if(i == i3)
+                return data[i][elementCountTill[i1] + i2];
+        
+        assert(0);
+    }
+
+    /++
+    Index operation of the form gp[["Grp Index"], ["Row-Index"], ["Column-Index"]]
+    +/
+    auto opIndex(T, U, V)(T i1, U i2, V i3)
+        if((is(T == string) || is(T == string[]))
+        && (is(U == string) || is(U == string[]))
+        && (is(V == string) || is(V == string[])))
+    {
+        int[3] pos;
+        static if(is(T == string))
+            pos[0] = getGroupPosition([i1]);
+        else
+            pos[0] = getGroupPosition(i1);
+        assert(pos[0] > -1, "Index out of bound");
+
+        static if(is(U == string))
+            pos[1] = positionInGroup!(0)([i2], pos[0]);
+        else
+            pos[1] = positionInGroup!(0)(i2, pos[0]);
+        assert(pos[1] > -1, "Index out of bound");
+
+        static if(is(V == string))
+            pos[2] = positionInGroup!(1)([i3], pos[0]);
+        else
+            pos[2] = positionInGroup!(1)(i3, pos[0]);
+        assert(pos[2] > -1, "Index out of bound");
+
+        static foreach(i; 0 .. GrpRowType.length)
+            if(i == pos[2])
+                return data[i][elementCountTill[pos[0]] + pos[1]];
+
+        assert(0);
+    }
+
+    /++
+    opIndex that returns Axis for column/row binary operations
+    +/
+    auto opIndex(Args...)(Args args)
```

Comment:
It seems the above to overloads of opIndex should be private and called something else, because there will be a collision in the overload resolution otherwise.

Reply:
Fixed this in both Group and DataFrame. Thanks for spotting this :+1: 
Addressed in 6517b4a